### PR TITLE
[sharing] Fix return type of `shareAsync`

### PR DESCRIPTION
**Pull request by barthap**
# Why

I needed to do something simple and relaxing 🙈 

Resolves ENG-4964 - see details in linear comment below

# How

- Changed the native implementations to resolve their promises with null/nil. Web is already ok
- Changed the return type of `shareAsync` to `Promise<void>`
- Regenerated docs

# Test Plan

none

### CODE REVIEW

File: packages/expo-sharing/CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,10 @@
 
 ### 🐛 Bug fixes
 
+### ⚠️ Notices
+
+- Changed the return type of the `shareAsync` function. The promise now resolves to `void` instead of an empty object. ([#18019](https://github.com/expo/expo/pull/18019) by [@barthap](https://github.com/barthap))
+
```
Comment (bbarthec) [inline packages/expo-sharing/CHANGELOG.md:14]: It's a breaking change IMO 🤔

**Review by bbarthec — CHANGES_REQUESTED** [review]
It looks good, but move CHANGELOG entry to `Breaking changes` section 👀

File: packages/expo-sharing/CHANGELOG.md (hunk 1 above)
Comment (barthap) [reply packages/expo-sharing/CHANGELOG.md:14]: It's unfortunate that such a small change would be marked as breaking 😞 I doubt that anybody ever used the return value.

Comment (bbarthec) [reply packages/expo-sharing/CHANGELOG.md:14]: Scope of a change is not a problem, but is it a breaking change indeed (API changed right?), so we need to be consistent in categorizing PRs under proper section.

**Review by bbarthec — APPROVED** [review]

**Pull request merged**